sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_pipeline_job_helpers.py

```diff
@@ -103,21 +111,35 @@ def from_dict_to_rest_io(
             # Add casting as sometimes we got value like 1(int)
             io_value = str(val.get("value", ""))
             io_mode = val.get("mode", None)
+            io_name = val.get("name", None)
+            io_version = val.get("version", None)
             if any([re.match(item, io_value) for item in io_binding_regex_list]):
+                io_bindings.update({key: {"path": io_value}})
+                # add mode to literal value for binding input
                 if io_mode:
                     # deal with dirty mode data submitted before
                     if io_mode in DIRTY_MODE_MAPPING:
                         io_mode = DIRTY_MODE_MAPPING[io_mode]
                         val["mode"] = io_mode
-                    # add mode to literal value for binding input
-                    io_bindings.update({key: {"path": io_value}})
                     if io_mode in OUTPUT_MOUNT_MAPPING_FROM_REST:
                         io_bindings[key].update({"mode": OUTPUT_MOUNT_MAPPING_FROM_REST[io_mode]})
                     else:
                         io_bindings[key].update({"mode": INPUT_MOUNT_MAPPING_FROM_REST[io_mode]})
-                else:
+                # add name and version for binding input
+                if io_name or io_version:
+                    assert rest_object_class.__name__ == 'JobOutput'
```

Comment:
why do we have assert here?

Reply:
I think further code may add name or version attribute for JobInput? In this case this assert can catch the situation and help modify code.

Reply:
I add related comments